airflow/providers/hashicorp/hooks/vault.py

```diff
@@ -0,0 +1,546 @@
+    :type role_id: str
+    :param kubernetes_role: Role for Authentication (for ``kubernetes`` auth_type)
+    :type kubernetes_role: str
+    :param kubernetes_jwt_path: Path for kubernetes jwt token (for ``kubernetes`` auth_type, default:
+        ``/var/run/secrets/kubernetes.io/serviceaccount/token``)
+    :type kubernetes_jwt_path: str
+    :param gcp_key_path: Path to GCP Credential JSON file (for ``gcp`` auth_type)
+    :type gcp_key_path: str
+    :param gcp_scopes: Comma-separated string containing GCP scopes (for ``gcp`` auth_type)
+    :type gcp_scopes: str
+
+    """
+    def __init__(  # pylint: disable=too-many-arguments
+        self,
+        vault_conn_id: str,
+        auth_type: Optional[str] = None,
+        mount_point: Optional[str] = None,
+        kv_engine_version: Optional[int] = None,
+        token: Optional[str] = None,
+        role_id: Optional[str] = None,
+        kubernetes_role: Optional[str] = None,
+        kubernetes_jwt_path: Optional[str] = None,
+        gcp_key_path: Optional[str] = None,
+        gcp_scopes: Optional[str] = None,
+    ):
+        super().__init__()
+        self.connection = self.get_connection(vault_conn_id)
+
+        if not auth_type:
+            auth_type = self.connection.extra_dejson.get('auth_type') or "token"
+
+        if not mount_point:
+            mount_point = self.connection.extra_dejson.get('mount_point')
+        if not mount_point:
+            mount_point = 'secret'
+
+        if not kv_engine_version:
+            conn_version = self.connection.extra_dejson.get("kv_engine_version")
+            try:
+                kv_engine_version = int(conn_version) if conn_version else DEFAULT_KV_ENGINE_VERSION
+            except ValueError:
+                raise VaultError(f"The version is not an int: {conn_version}. ")
+
+        if auth_type in ["approle"]:
+            if not role_id:
+                role_id = self.connection.extra_dejson.get('role_id')
+
+        if auth_type in ['github', "token"]:
+            if not token:
+                token = self.connection.extra_dejson.get('token')
+        gcp_key_path, gcp_scopes = \
+            self._get_gcp_parameters_from_connection(gcp_key_path, gcp_scopes) \
+            if auth_type == 'gcp' else (None, None)
+        kubernetes_jwt_path, kubernetes_role = \
+            self._get_kubernetes_parameters_from_connection(kubernetes_jwt_path, kubernetes_role) \
+            if auth_type == 'kubernetes' else (None, None)
+
+        url = f"{self.connection.schema}://{self.connection.host}"
+        if self.connection.port:
+            url += f":{self.connection.port}"
+
+        self.vault_client = _VaultClient(
+            url=url,
+            auth_type=auth_type,
+            mount_point=mount_point,
+            kv_engine_version=kv_engine_version,
+            token=token,
+            username=self.connection.login,
+            password=self.connection.password,
+            secret_id=self.connection.password,
```

Comment:
```suggestion
            secret_id=self.secret_id,
```

I think?

Reply:
In the hook I mapped secret_id from the password (this is only used in approle method). it makes more sense to use password from the connection - mostly because it is replaced with *********  in the UI, so storing that in extras as "secret_id" is not a good idea I think.

This is explained in the Hook's docstring
```
    Login/Password are used as credentials:

        * approle: password -> secret_id
        * ldap: login -> username,   password -> password
        * userpass: login -> username, password -> password

```

Reply:
Ah gotcha. :+1: